src/test/scala/strawman/collection/test/TraverseTest.scala

```diff
@@ -29,14 +30,21 @@ class TraverseTest {
   def traverseTest: Unit = {
 
     val xs1 = immutable.List(Some(1), None, Some(2))
-    optionSequence(xs1)(immutable.List.canBuild[Int]) // TODO Remove explicit CanBuild parameter after https://issues.scala-lang.org/browse/SI-10081 is fixed?
+    val o1 = optionSequence(xs1)
+    val o1t: Option[immutable.List[Int]] = o1
 
     val xs2 = immutable.TreeSet(Some("foo"), Some("bar"), None)
-    optionSequence(xs2)(immutable.TreeSet.canBuild[String])
+    val o2 = optionSequence(xs2)
```

Comment:
No explicit factory required, even for constrained collections. Both the previous implementation here and my old attempt in https://github.com/scala/collection-strawman/pull/39 needed it.